app.go

```diff
@@ -340,6 +351,16 @@ func (h *Headscale) httpAuthenticationMiddleware(c *gin.Context) {
 	// c.Next()
 }
 
+// ensureUnixSocketIsAbsent will check if the given path for headscales unix socket is clear
+// and will remove it if it is not.
+func (h *Headscale) ensureUnixSocketIsAbsent() error {
+	// File does not exist, all fine
+	if _, err := os.Stat(h.cfg.UnixSocket); errors.Is(err, os.ErrNotExist) {
+		return nil
+	}
+	return os.Remove(h.cfg.UnixSocket)
```

Comment:
I think it will crash the other one. I am a bit conflicted trying to fix that, because at some point I think it must become the responsibility of a init system like systemd 🤔

Reply:
Mmmm. Kinda agreed!